packages/router/router.ts

```diff
@@ -3672,18 +3722,50 @@ function createClientSideRequest(
   let init: RequestInit = { signal };
 
   if (submission && isMutationMethod(submission.formMethod)) {
-    let { formMethod, formEncType, formData } = submission;
+    let { formMethod, formEncType, formData, payload } = submission;
     // Didn't think we needed this but it turns out unlike other methods, patch
     // won't be properly normalized to uppercase and results in a 405 error.
     // See: https://fetch.spec.whatwg.org/#concept-method
     init.method = formMethod.toUpperCase();
-    init.body =
-      formEncType === "application/x-www-form-urlencoded"
-        ? convertFormDataToSearchParams(formData)
-        : formData;
+
+    if (formData) {
+      if (formEncType === "application/x-www-form-urlencoded") {
+        // Content-Type is inferred (https://fetch.spec.whatwg.org/#dom-request)
+        init.body = convertFormDataToSearchParams(formData);
+      } else {
+        // Content-Type is inferred (https://fetch.spec.whatwg.org/#dom-request)
+        init.body = formData;
+      }
+    } else if (payload != null) {
+      if (formEncType === "application/x-www-form-urlencoded") {
+        let payloadFormData = new FormData();
+
+        if (payload instanceof URLSearchParams) {
+          for (let [name, value] of payload) {
+            payloadFormData.append(name, value);
+          }
+        } else if (payload != null) {
+          for (let name of Object.keys(payload)) {
+            // @ts-expect-error
+            payloadFormData.append(name, payload[name]);
+          }
+        }
+
+        // Content-Type is inferred (https://fetch.spec.whatwg.org/#dom-request)
+        init.body = convertFormDataToSearchParams(payloadFormData);
+      } else if (formEncType === "application/json") {
+        init.headers = new Headers({ "Content-Type": formEncType });
+        init.body = JSON.stringify(payload);
+      } else if (formEncType === "text/plain") {
+        init.headers = new Headers({
+          "Content-Type": formEncType,
+        });
+        init.body =
+          typeof payload === "string" ? payload : JSON.stringify(payload);
+      }
+    }
```

Comment:
Serialize the incoming `formData` or `payload` onto the `request` if we know how

Reply:
@ryanflorence Tagging for review - new router serialization based on content-type